modules/ide-integration/README.md

```diff
@@ -1,228 +1,292 @@
 
-  ```
-  127.0.0.1 airavata.host
-  ```
+```bash
+cd modules/ide-integration/src/main/containers
+docker-compose up -d
+```
 
-* Go to src/main/containers directory and run 
+**Services Started:**
+- 🗄️ **MySQL Database**
+- 🔐 **Keycloak** (Authentication)
+- 📨 **Apache Kafka** (Messaging)
+- 🐰 **RabbitMQ** (Message Queue)
+- 🔒 **SSHD Server** (Secure connections)
 
-  ```
-  docker-compose up
-  ```
+### 5️⃣ Initialize Database
+
+Apply database migrations:
+
+```bash
+cd modules/ide-integration/src/main/containers
+cat ./database_scripts/init/*-migrations.sql | docker exec -i containers-db-1 mysql -p123456
+```
 
-* Apply any database migrations. Go to src/main/containers directory and run
+⏳ **Wait for all services to be ready** (usually 2-3 minutes)
 
-  ```
-  cat ./database_scripts/init/*-migrations.sql | docker exec -i containers-db-1 mysql -p123456
-  ```
+## 🖥️ Starting Airavata Components
 
-* Wait until all the services come up. This will initialize all utilities required to start Airavata server
+### 6️⃣ Start API Server
 
-### Starting API Server
+1. **Navigate to:** `org.apache.airavata.ide.integration.APIServerStarter`
+2. **Right-click** in the editor
+3. **Select:** `Run 'APIServerStarter.main()'`
 
-#### Note: For JDK 11+
-you have to add ``--add-opens java.base/java.lang=ALL-UNNAMED`` as a JVM argument
+> 💡 **JDK 17+ Note:** Add this JVM argument in your run configuration:
+> ```
+> --add-opens java.base/java.lang=ALL-UNNAMED
+> ```
 
-* Go to org.apache.airavata.ide.integration.APIServerStarter class and right click on the editor and click Run option. This will start Airavata server
+### 7️⃣ Start Job Execution Engine
 
-### Starting Job Execution Engine
+1. **Navigate to:** `org.apache.airavata.ide.integration.JobEngineStarter`
+2. **Right-click** and select **Run**
 
-* Go to org.apache.airavata.ide.integration.JobEngineStarter class and right click on the editor and click Run option. 
-This will start all components of Job Execution Engine including Helix Controller, Helix Participant, Pre Workflow Manager and 
-Post Workflow Manager
+**Components Started:**
+- 🔄 Helix Controller
+- 👥 Helix Participant  
+- ⚙️ Pre Workflow Manager
+- 📋 Post Workflow Manager
 
-### Starting Job Monitoring components
+### 8️⃣ Start Job Monitoring
 
-* This will start the Email Based Job Monitoring agent. Before starting this, you have to create a new gmail account by going to 
-https://accounts.google.com/signup
+**Setup Email Monitor (One-time setup):**
 
-* Once the account is created, turn on 2-Step Verification and create an App Password (Use the type "Other" from the App type selection and give the name as "Airavata")
-https://myaccount.google.com/security
+1. **Create Gmail Account:** [accounts.google.com/signup](https://accounts.google.com/signup)
+2. **Enable 2-Step Verification:** [myaccount.google.com/security](https://myaccount.google.com/security)
+3. **Generate App Password:**
```

Comment:
mention that the user has to navigate to `https://myaccount.google.com/apppasswords` to create an app pw